crypto/crypto/src/ed25519.rs

```diff
@@ -108,8 +108,12 @@ impl Ed25519Signature {
         }
     }
 
-    /// Check for correct size and malleability issues.
-    /// This method ensures s is of canonical form and R does not lie on a small group.
+    /// Check for correct size and third-party based malleability issues.
+    /// This method ensures s is of canonical form.
+    /// Regarding the R part of the signature, this method does not check if it lies in a small
```

Comment:
I think this comment does not convey the essential information: we do have to (and perform) a check on s. Would it be possible to shorten it, and make it more local to the definition of the relevant bit slice for R?